### PR TITLE
Add logic for preserving attachments in `NewEmailFromReader`

### PR DESCRIPTION
**Pull request by danielsig727**
## why
To make the parsed `Email` object contains attachments correctly, which previously all attachments are disregarded. :(

## How
Attachment parts are actually parsed as `part` objects already, but they're not added into the `Email` object. Added helper function to construct the attachment. Also, tests are added.

### CODE REVIEW

File: email.go [hunk 1]
```diff
@@ -126,6 +127,12 @@ func NewEmailFromReader(r io.Reader) (*Email, error) {
 			return e, err
 		}
 		switch {
+		case strings.Contains(p.header.Get("Content-Disposition"), "attachment"):
+			attachment, err := parseAttachment(p)
+			if err != nil {
+				return e, err
```
Comment (kkc) [inline email.go:133]: shouldn't it be `nil, err`?

File: email.go [hunk 2]
```diff
@@ -135,6 +142,21 @@ func NewEmailFromReader(r io.Reader) (*Email, error) {
 	return e, nil
 }
 
+// parseAttachment constructs an Attachment object from the given parsed part object
+func parseAttachment(part *part) (*Attachment, error) {
+	attachment := &Attachment{
+		Header: part.header,
+		Content: part.body,
+	}
+	matches := regexp.MustCompile(`filename=\"?([^\"]+)\"`).FindAllStringSubmatch(
```
Comment (kkc) [inline email.go:152]: can we add some comments for this regex?

File: email.go (hunk 1 above)
Comment (danielsig727) [reply email.go:133]: I agree... but other parts of this function is written as `e, err` by the author... 🤔
I'm just following the convention

**Review by kkc — APPROVED** [review]
LGTM

**Pull request merged**